### PR TITLE
extend compatibility of array deserialization in lookers

### PR DESCRIPTION
**Pull request by tom-vx51**
## What changes are proposed in this pull request?

* Extend array deserialization in lookers to include additional fields
* Fix bug where `this.targets` can be undefined in `Segmentation`, causing a crash

## How is this patch tested? If it is not, please explain why.

local app

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Improved error handling in the Segmentation overlay for better target validation.
	- Enhanced deserialization process for masks and maps, accommodating various input formats.

- **Bug Fixes**
	- Fixed potential out-of-bounds access in the Segmentation overlay's target retrieval logic.

### CODE REVIEW

File: app/packages/looker/src/worker/deserializer.ts [hunk 1]
```diff
@@ -2,8 +2,15 @@ import { deserialize } from "../numpy";
...
+    let serializedMask: string;
     if (typeof label?.mask === "string") {
-      const data = deserialize(label.mask);
+      serializedMask = label.mask;
+    } else if (typeof label?.mask?.$binary?.base64 === "string") {
+      serializedMask = label.mask.$binary.base64;
+    }
+
+    if (serializedMask) {
+      const data = deserialize(serializedMask);
```
Comment (jleven) [inline app/packages/looker/src/worker/deserializer.ts:13]: I wouldn't be upset if you removed all the code duplication.

```typescript
    const data = deserializeProp(label, "mask");
    if (data) {
...

function deserializeProp(label, prop) {
  let serializedMask: string | null = null;
  if (typeof label?.[prop] === "string") {
      serializedMask = label[prop];
    } else if (typeof label?.[prop]?.$binary?.base64 === "string") {
      serializedMask = label[prop].$binary.base64;
    }
  }
  if (!serializedMask) {
    return null;
  }
  return deserialize(serializedMask);
}
```

Comment (tom-vx51) [reply app/packages/looker/src/worker/deserializer.ts:13]: yeah agreed - I started to and then it was more code than it was worth so I got lazy. I'll refactor.

**Review by sashankaryal — APPROVED** [review]
lgtm! 🚀

File: app/packages/looker/src/worker/deserializer.ts (hunk 1 above)
Comment (jleven) [reply app/packages/looker/src/worker/deserializer.ts]: Thanks!

**Pull request merged**